src/vs/code/node/cli.ts

```diff
@@ -205,10 +211,22 @@ export async function main(argv: string[]): Promise<any> {
 			// - the wait marker file has been deleted (e.g. when closing the editor)
 			// - the launched process terminates (e.g. due to a crash)
 			processCallbacks.push(async child => {
+				let childExitPromise = Event.toPromise(Event.fromNodeEventEmitter(child, 'exit'));
```

Comment:
I would document why the `childExitPromise` is different on Windows/Linux vs Mac, because otherwise it is impossible to understand the logic here. And I would do an `if/else` based on the platforms and not call `let childExitPromise = Event.toPromise(Event.fromNodeEventEmitter(child, 'exit'));` on macOS.